modules/core/04-channel/keeper/packet_test.go

```diff
@@ -730,6 +730,7 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 		packet types.Packet
 		ack    = ibcmock.MockAcknowledgement
 
+		assertFn   func()
 		channelCap *capabilitytypes.Capability
 		expError   *errorsmod.Error
```

Comment:
I guess this `expError` was previously added as an afterthought/temporary change to the tests.

I'd like to refactor these tests to use either `expError` or `expResult` (func) as part of the testcase structure. I can create an issue and do this in a follow up if we are aligned on that. 

I think it would make testing here more flexible, we can do expected error checking in an `expResult` func also, as well as expected posterior state

Reply:
I like the function expResult exactly because of the ability to verify posterior state so ++

Reply:
created #4433 and assigned to myself